client/src/main/scala/drt/client/components/StaffingUtil.scala

```diff
@@ -0,0 +1,113 @@
+import uk.gov.homeoffice.drt.time.SDateLike
+
+object StaffingUtil {
+  def navigationDates(viewingDate: SDateLike, isWeekly: Boolean, isDaily: Boolean, now: () => SDateLike): (SDateLike, SDateLike) = {
+    val currentMonthStart = SDate.firstDayOfMonth(now())
+    val adjustedViewingDate = if (viewingDate.millisSinceEpoch < currentMonthStart.millisSinceEpoch) {
+      currentMonthStart
+    } else {
+      viewingDate
+    }
+
+    if (isWeekly) {
+      val previousWeekDate = {
+        val firstDayOfWeek = SDate.firstDayOfWeek(viewingDate)
+        val potentialPreviousWeekDate = if (firstDayOfWeek.millisSinceEpoch < currentMonthStart.millisSinceEpoch) {
+          adjustedViewingDate
+        } else adjustedViewingDate.addDays(-7)
+        potentialPreviousWeekDate
+      }
+      val nextWeekDate = {
+        val currentMonthStart = SDate.firstDayOfMonth(now())
+        val lastDayOfFifthMonth = SDate.lastDayOfMonth(currentMonthStart.addMonths(5))
+        val lastDayOfWeek = SDate.lastDayOfWeek(viewingDate)
+
+        if (lastDayOfWeek.millisSinceEpoch > lastDayOfFifthMonth.millisSinceEpoch) {
+          adjustedViewingDate
+        } else {
+          adjustedViewingDate.addDays(7)
+        }
+      }
+      (previousWeekDate, nextWeekDate)
+    } else if (isDaily) {
+      val previousDayDate = adjustedViewingDate.addDays(-1)
+      val nextDayDate = {
+        val currentMonthStart = SDate.firstDayOfMonth(now())
+        val lastDayOfFifthMonth = SDate.lastDayOfMonth(currentMonthStart.addMonths(5))
+        if (adjustedViewingDate.millisSinceEpoch == lastDayOfFifthMonth.millisSinceEpoch) {
+          adjustedViewingDate
+        } else {
+          adjustedViewingDate.addDays(1)
+        }
+      }
+      val isFirstDayOfCurrentMonth = adjustedViewingDate.millisSinceEpoch == currentMonthStart.millisSinceEpoch
+      if (isFirstDayOfCurrentMonth)
+        (currentMonthStart, nextDayDate)
+      else
+        (previousDayDate, nextDayDate)
+    } else {
+      val previousMonthDate = adjustedViewingDate.addMonths(-1)
+      val nextMonthDate = adjustedViewingDate.addMonths(1)
+      val sixMonthsFromNow = currentMonthStart.addMonths(5)
+      val finalNextMonthDate = if (nextMonthDate.millisSinceEpoch > sixMonthsFromNow.millisSinceEpoch)
+        sixMonthsFromNow
+      else
+        nextMonthDate
+      if (previousMonthDate.millisSinceEpoch < currentMonthStart.millisSinceEpoch) {
+        (currentMonthStart, finalNextMonthDate)
+      } else
+        (previousMonthDate, finalNextMonthDate)
+    }
+  }
+
+  def consecutiveDayForWeek(viewingDate: SDateLike): Seq[(SDateLike, String)] = {
+    val startOfWeek = SDate.firstDayOfWeek(viewingDate)
+    val endOfWeek = SDate.lastDayOfWeek(viewingDate)
+
+    val days = (endOfWeek.millisSinceEpoch - startOfWeek.millisSinceEpoch) / (1000 * 60 * 60 * 24) + 1
+    daysOfWeek(startOfWeek, days.toInt)
+  }
+
+  def consecutiveDay(viewingDate: SDateLike): Seq[(SDateLike, String)] = {
```

Comment:
I don't think this function is very clear. It's not a consecutive day if there's only one day. All it does is return the same date and adds the name of the day for that date. We either need a clearer name or maybe inline the code where it's being called from, which I think is the more sensible option